syntax = "proto3";
option optimize_for = LITE_RUNTIME;

package Jde.Http.Proto;
import "FromServer.proto";
import "FromClient.proto";

message Echo{
	uint32 request_id=1;
	string echo_text=2;
}

message TestFromServerUnion{
	oneof Value{
		FromServer.Ack ack=1;
//		FromServer.Complete complete=2;
		FromServer.Exception exception=3;
		Echo echo=4;
	}
}

message TestFromServer{
	repeated TestFromServerUnion messages = 1;
}

message CloseServerSide{
};

message BadTransmissionServer{
};

message TestFromClientUnion{
	oneof Value{
		FromClient.Connect connect=1;
		Echo echo=2;
		CloseServerSide close_server_side=3;
		BadTransmissionServer bad_transmission_server=4;
	}
}

message TestFromClient{
	repeated TestFromClientUnion messages = 1;
}